tests/test_llm_interface.py

```diff
@@ -0,0 +1,105 @@
+        return responses
+
+
+def test_large_language_model():
+    """Tests for LargeLanguageModel()."""
+    cache_dir = "_fake_llm_cache_dir"
+    utils.reset_flags({
+        "llm_cache_dir": cache_dir,
+        "llm_use_cache_only": False,
+        "llm_openai_max_response_tokens": 700
+    })
+    # Remove the fake cache dir in case it's lying around from old tests.
+    shutil.rmtree(cache_dir, ignore_errors=True)
+    # Query a dummy LLM.
+    llm = _DummyLLM()
+    assert llm.get_id() == "dummy"
+    responses = llm.sample_completions("Hello world!", 0.5, 123, 3)
+    completions = [r.text for r in responses]
+    expected_completion = "Prompt was: Hello world!. Seed: 123. Temp: 0.5."
+    assert completions == [expected_completion] * 3
+    # Query it again, covering the case where we load from disk.
+    responses = llm.sample_completions("Hello world!", 0.5, 123, 3)
+    completions = [r.text for r in responses]
+    assert completions == [expected_completion] * 3
+    # Query with temperature 0.
+    responses = llm.sample_completions("Hello world!", 0.0, 123, 3)
+    completions = [r.text for r in responses]
+    expected_completion = "Prompt was: Hello world!. Seed: 123. Temp: 0.0."
+    assert completions == [expected_completion] * 3
+    # Clean up the cache dir.
+    shutil.rmtree(cache_dir)
+    # Test llm_use_cache_only.
+    utils.reset_flags({
+        "llm_cache_dir": cache_dir,
+        "llm_use_cache_only": True,
+        "llm_openai_max_response_tokens": 700
+    })
+    with pytest.raises(ValueError) as e:
+        completions = llm.sample_completions("Hello world!", 0.5, 123, 3)
+    assert "No cached response found for LLM prompt." in str(e)
+
+
+def test_openai_llm():
+    """Tests for OpenAILLM()."""
+    cache_dir = "_fake_llm_cache_dir"
+    utils.reset_flags({
+        "llm_cache_dir": cache_dir,
+        "llm_use_cache_only": False,
+        "llm_openai_max_response_tokens": 700
+    })
+    if "OPENAI_API_KEY" not in os.environ:
+        os.environ["OPENAI_API_KEY"] = "dummy API key"
+    # Create an OpenAILLM with the curie model.
+    llm = OpenAILLM("text-curie-001")
+    assert llm.get_id() == "openai-text-curie-001"
+    # Test _raw_to_llm_response().
+    raw_response = {
+        "text": "Hello world",
+        "logprobs": {
+            "tokens": ["Hello", "world"],
+            "token_logprobs": [-1.0, -2.0]
+        }
+    }
+    llm_response = llm._raw_to_llm_response(raw_response)  # pylint: disable=protected-access
+    assert llm_response.text == "Hello world"
+    assert llm_response.tokens == ["Hello", "world"]
+    assert llm_response.token_logprobs == [-1.0, -2.0]
+    # Uncomment this to test manually, but do NOT uncomment in master, because
+    # each query costs money.
+    # completions = llm.sample_completions("Hello", 0.5, 123, 2)
+    # assert len(completions) == 2
+    # completions2 = llm.sample_completions("Hello", 0.5, 123, 2)
+    # assert completions == completions2
```

Comment:
wont these completions typically not equal each other since we have a non zero temperature here?

Reply:
they should equal each other because of caching